tailcfg/tailcfg.go

```diff
@@ -873,6 +874,10 @@ type DNSConfig struct {
 
 	// PerDomain is not set by the control server, and does nothing.
 	PerDomain bool `json:",omitempty"`
+
+	// SetDNSDomains are the set of DNS names that are permitted to
```

Comment:
I'm not sure about the field name. SetDNSDomains sounds like an imperative command for the client to set some DNS domains, not "these are the domains you can choose to request stuff on". We also don't guarantee that you can set whatever records you want on these names, it's just a hint that "if you're trying to do LE, these are the ones that are likely to succeed".

I don't have better ideas off the top of my head, the DNS record-setting stuff in tailcfg is so generic that it's hard to come up with reasonable names :/

Reply:
ValidDNSCertDomain ?

Reply:
ValidCertDomains?

Reply:
What's the motivation for this field? Does it get set by the tailscale administrator to limit which machines may set TXT records, or does it limit which "sub-names" are allowed (eg: `_acme-challenge`), or is it both? Should it allow wildcards, eg: `_acme-challenge.*` (to allow any machine to use LE), or `*.some.awesome.machine` (which can set a TXT record for _any_ sub-name)

Reply:
@normanr, we haven't decided what subdomain structure we're going to use for LetsEncrypt certs. There are various concerns about customer hostnames/emails leaking to CT logs if we do it with our current MagicDNS names. So we might make this opt-in with separate hostnames. This is giving us a place to advertise that policy from the server to the client later, so we can update it server side and not require client updates.

Reply:
Decision from @danderson: just `CertDomains`.